m3u8/writer.go

```diff
@@ -688,16 +708,21 @@ func NewMediaPlaylist(winsize uint, capacity uint) (*MediaPlaylist, error) {
 	p.ver = minVer
 	p.winsize = winsize
 	p.capacity = capacity
-	p.Segments = make([]*MediaSegment, capacity)
 	p.writePrecision = DefaultFloatPrecision
 	return p, nil
 }
 
-// ReleasePlaylist returns buffer and segment slice to pool for reuse
-// Do not use the playlist after this
+// ReleasePlaylist returns buffer and segment slice to pool for reuse. Do not use the playlist after this
 func (p *MediaPlaylist) ReleasePlaylist() {
-	putBuffer(p.buf)
-	putSegmentSlice(p.Segments)
+	putBuffer(&p.buf)
+	putSegmentSlice(&p.Segments)
+}
+
+// ReleaseSegments returns all segments to pool for reuse. Do not use the segments after this.
+func (p *MediaPlaylist) ReleaseSegments() {
```

Comment:
I think this shouldn't be part of the public API for the playlist. Maybe you could keep it private and then call in in `ReleasePlaylist()`